src/virtual_machine.rs

```diff
@@ -34,6 +77,113 @@ impl VM {
             rcount: 0,
         }
     }
+
+    fn decode_instruction(instruction: u16) -> Opcode {
+        // Removes the arguments from the instruction, leaving only the operator
+        let op = instruction >> ARGSIZE;
+        // Removes the operator from the instruction, leaving only the arguments
+        let args = instruction & ARGUMENTMASK;
+
+        // NOTE: Whilst writing constants inside a code block
+        // is not very orthodox, I believe it remains a good
+        // balance between avoiding magic numbers and keeping
+        // the logic "local".
+        match op {
+            // BR
+            0b0000 => {
+                todo!()
+            }
+            // ADD
+            0b0001 => {
+                const ADDMODEPOSITION: u16 = 5;
+                const ADDMODEFLAG: u16 = 0b0000000000100000;
+                let mode = (args & ADDMODEFLAG) >> ADDMODEPOSITION;
```

Comment:
Why not use the literal directly in the line of code?
`let mode = (args & 0b0000000000100000) >> 5;`